src/cogent3/app/data_store_new.py

```diff
@@ -272,53 +287,150 @@ def completed(self) -> list[DataMember]:
 
     @property
     def not_completed(self) -> list[DataMember]:
-        ic_dir = self.source / _NOT_COMPLETED_TABLE
+        nc_dir = self.source / _NOT_COMPLETED_TABLE
         return (
             [
                 DataMember(self, Path(_NOT_COMPLETED_TABLE) / m.name)
-                for m in ic_dir.glob("*.json")
+                for m in nc_dir.glob("*.json")
             ]
-            if ic_dir.exists()
+            if nc_dir.exists()
             else []
         )
 
     def write(self, unique_id: str, data: str) -> None:
         super().write(unique_id, data)
+        # check suffix compatible with this datastore
+        suf, cmp = get_format_suffixes(unique_id)
+        self.suffix = self.suffix or suf
+        if suf != self.suffix:
+            unique_id = f"{unique_id}.{self.suffix}"
         with open_(self.source / unique_id, mode="w") as out:
             out.write(data)
-            if self._md5 and isinstance(data, str):
-                self._checksums[unique_id] = get_text_hexdigest(data)
+        md5 = get_text_hexdigest(data)
+        unique_id = unique_id.replace(self.suffix, "txt")
+        unique_id = unique_id if cmp is None else unique_id.replace(f".{cmp}", "")
+        with open_(self.source / _MD5_TABLE / unique_id, mode="w") as out:
+            out.write(md5)
 
     def write_not_completed(self, unique_id: str, data: str) -> None:
         super().write(unique_id, data)
+        # check suffix compatible with this datastore
+        suf, cmp = get_format_suffixes(unique_id)
+        self.suffix = self.suffix or suf
+        if suf != self.suffix:
```

Comment:
should not be using `self.suffix` for not completed, should be using ".json", but see suggested refactor